src/learning/ProfileInitializer.cpp

```diff
@@ -0,0 +1,214 @@
+    catAbove = cat.getCategoryRank();
+    catBelow = catAbove - 1;
+  } else {
+    catBelow = cat.getCategoryRank();
+    catAbove = catBelow + 1;
+  }
+  // std::cout << altPerformance_.getPerformanceTable();
+  for (std::vector<Perf> vPerf : altPerformance_.getPerformanceTable()) {
+    if (altPerformance_.getAlternativeAssignment(vPerf[0].getName())
+                .getCategoryRank() == catBelow ||
+        altPerformance_.getAlternativeAssignment(vPerf[0].getName())
+                .getCategoryRank() == catAbove) {
+      for (Perf p : vPerf) {
+        if (p.getCrit() == crit.getId()) {
+          candidates.push_back(p);
+          break;
+        }
+      }
+    }
+  }
+  return candidates;
+}
+
+float ProfileInitializer::weightedProbability(
+    const Perf perfAlt, const Criterion &crit, const Category &catAbove,
+    const Category &catBelow, const int nbCategories,
+    const std::vector<float> &catFrequency, std::vector<Perf> &candidates,
+    float delta) {
+  // creating imaginary profile performance for criterion crit
+  std::string critId = crit.getId();
+  float imaginaryProfilePerformance = perfAlt.getValue() + delta;
+
+  // Creating 2 int that will count the number of correctly classified
+  // alternatives for criterion crit for a profile performance of
+  // imaginaryProfilePerformance.
+  int catAboveCounter = 0;
+  int catBelowCounter = 0;
+
+  // candidates are computed thanks to the method
+  // ProfileInitializer::getProfilePerformanceCandidates(crit, catBelow,
+  // nbCategories);
+  for (Perf can : candidates) {
+    // if the performance of the candidate is higher than the profile
+    // performance and that the alternatives category is catAbove then we have
+    // correctly classified it
+    if (can.getValue() > imaginaryProfilePerformance and
+        altPerformance_.getAlternativeAssignment(can.getName()) == catAbove) {
+      ++catAboveCounter;
+      // if the performance of the candidate is lower than the profile
+      // performance and that the alternatives category is catBelow then we
+      //  have correctly classified it
+    } else if (can.getValue() < imaginaryProfilePerformance and
+               altPerformance_.getAlternativeAssignment(can.getName()) ==
+                   catBelow) {
+      ++catBelowCounter;
+    }
+  }
+  float proba;
+  proba = catAboveCounter / catFrequency[catAbove.getCategoryRank()] +
+          catBelowCounter / catFrequency[catBelow.getCategoryRank()];
+  return proba;
+}
+
+std::vector<Perf> ProfileInitializer::initializeProfilePerformance(
+    const Criterion &crit, Categories &categories,
+    const std::vector<float> &catFre) {
+  std::vector<float> categoryLimits;
+  int nbCategories = categories.getNumberCategories();
+
+  for (int i = 0; i < nbCategories - 1; i++) {
+    std::vector<float> altProba;
+    std::vector<Perf> candidates =
+        ProfileInitializer::getProfilePerformanceCandidates(crit, categories[i],
```

Comment:
You might want to double check that, but I think if the method is not static, it should be call with `this->`.
However, we constructed the algorithm to have our methods independent from the state, so that could be a good idea to put most of the method in static and call them as you did here!